ui/app/clinical/common/controllers/visitController.js

```diff
@@ -16,6 +16,24 @@ angular.module('bahmni.clinical')
             $scope.patientUuid = $stateParams.patientUuid;
             $scope.visitUuid = $stateParams.visitUuid;
             var tab = $stateParams.tab;
+            $scope.$on('observations.providers', function (event, observations) {
```

Comment:
do we need to use the $on?

Reply:
Yes Angshu this will update the provider when there is another visit in same encounter and that visit is started by different doctor in that case UI should be updated with provider who started the visit.So this would capture whenever there is change in provide
<img width="716" alt="Screenshot 2023-02-17 at 6 50 11 PM" src="https://user-images.githubusercontent.com/96411257/219664014-2d67e8b5-b397-4b66-9f56-ded62a156843.png">
<img width="1515" alt="Screenshot 2023-02-17 at 6 50 29 PM" src="https://user-images.githubusercontent.com/96411257/219664065-6b530522-f987-4b18-8c57-d20676871121.png">
<img width="1588" alt="Screenshot 2023-02-17 at 6 50 40 PM" src="https://user-images.githubusercontent.com/96411257/219664146-585005df-58d6-4900-9a35-7cc0090dd71c.png">